src/EnergyOrigin.PedersenCommitment/Commitment.cs

```diff
@@ -0,0 +1,42 @@
+using System.Numerics;
+
+namespace EnergyOrigin.PedersenCommitment;
+
+public record Commitment
+{
+    public BigInteger C { get; }
+    public Group Group { get; }
+
+    public Commitment(BigInteger c, Group group)
+    {
+        if (BigInteger.ModPow(c, group.q, group.p) != 1 % group.p) throw new InvalidDataException("C^q should be equal to 1 mod p");
+
+        C = c;
+        Group = group;
+    }
+
+    public static Commitment Create(Group group, BigInteger m, params BigInteger[] r)
+    {
+        var rSum = BigInteger.Zero;
+        foreach (var rEl in r)
+        {
+            rSum += rEl;
+        }
+
+        return new Commitment(BigInteger.ModPow(group.g, m, group.p) * BigInteger.ModPow(group.h, rSum, group.p) % group.p, group);
+    }
+
```

Comment:
Is it valuable to add a +(addition) and -(subtraction) operator as well ?
My notes says that they are the same as * / but using mod q.